Tests/MongoSwiftTests/ConnectionStringTests.swift

```diff
@@ -441,4 +441,49 @@ final class ConnectionStringTests: MongoSwiftTestCase {
             options: MongoClientOptions(localThresholdMS: tooLarge)
         )).to(throwError(errorType: MongoError.InvalidArgumentError.self))
     }
+
+    func testCompressionOptions() throws {
+        // zlib level validation
+        expect(try Compressor.zlib(level: -2)).to(throwError(errorType: MongoError.InvalidArgumentError.self))
```

Comment:
debated getting rid of the level label so you'd just do `.zlib(6)` etc. but I wasn't sure if the meaning of the value was obvious enough without it. thoughts?

Reply:
One way to alleviate this would be to define an enum with cases for `.disabled`, `.defaultLevel`, or `.level(Int)`. Then it'd look like `.zlib(.level(6))`.

Reply:
would this give the user the ability to do anything they can't already now?
given the override behavior, it seems to me that passing in both `compressors: []` and `compressors: [.zlib(.disabled))]` would have the same effect.

`.default` also seems somewhat contradictory to our usual practice of unspecified=default

Reply:
So default in this case really means "enabled with the default level", so maybe we make `level` argument an `Int? = nil` one. As far as `.disabled` goes, I was just translating the option 0 = disabled, but I agree that having it included doesn't really add much so we can probably just go back to `Int`.

Reply:
I think the label looks nice, just my two cents

Reply:
ya I'm good with keeping the label.

I used the static property `zlib` plus the throwing static method `zlib(level:)` so that users can just do `.zlib` and not `try .zlib()` when they aren't providing a level